src/units/units.cpp

```diff
@@ -0,0 +1,161 @@
+//========================================================================================
+// Athena++ astrophysical MHD code
+// Copyright(C) 2014 James M. Stone <jmstone@princeton.edu> and other code contributors
+// Licensed under the 3-clause BSD License, see LICENSE file for details
+//========================================================================================
+//! \file units.cpp
+//! \brief define unit class and physical constants
+
+// C headers
+
+// C++ headers
+#include <iostream>
+#include <sstream>    // stringstream
+#include <stdexcept> //throw exceptions
+
+// Athena++ headers
+#include "../athena.hpp"
+#include "../athena_arrays.hpp"
+#include "../parameter_input.hpp"
+#include "units.hpp"
+
+//========================================================================================
+//! \fn void Units::Units(ParameterInput *pin)
+//! \brief default unit constructor from the parameter input (default is for ISM problems)
+//!        temperature units are not set (due to the mu dependence)
+//========================================================================================
+Units::Units(ParameterInput *pin) :
+  unit_system(pin->GetOrAddString("units", "unit_system", "ism")) {
+  // define units for given unit system
+  // 1) ism unit system adopted in TIGRESS
+  //   (note the slight change in the density units 1.4271->1.4):
+  //     [rho] = 1.4 m_h/cm^3
+  //     [length] = pc
+  //     [velocity] = km/s
+  //   which gives MLT units
+  //     [mass] = 1.4 m_h*(pc/cm)^3
+  //     [length] = pc
+  //     [time] = (pc/km)*s
+  // 2) galaxy unit system (example)
+  //     [mass] = 1.4 m_h*(pc/cm)^3
+  //     [length] = kpc
+  //     [time] = Myr
+  // add other default units system here
+  // set private MLT unit variables here
+  if (unit_system.compare("ism") == 0) {
+    code_length_cgs_ = Constants::pc_cgs;
+    code_mass_cgs_ = 1.4*Constants::hydrogen_mass_cgs*CUBE(code_length_cgs_);
+    code_time_cgs_ = Constants::pc_cgs/Constants::km_s_cgs;
+  } else if (unit_system.compare("galaxy") == 0) {
+    code_length_cgs_ = Constants::kpc_cgs;
+    code_mass_cgs_ = 1.4*Constants::hydrogen_mass_cgs*CUBE(code_length_cgs_);
+    code_time_cgs_ = Constants::million_yr_cgs;
```

Comment:
It is better to minimize problem specific codes in the master branch. Technically this can be done in the input file for each problem (i.e., always "custom"). Is there any reason to have them hard coded?